pkgs/development/python-modules/pynvml/0001-locate-libnvidia-ml.so.1-on-NixOS.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/pynvml/nvml.py b/pynvml/nvml.py
+index 56d908f..1de0b97 100644
+--- a/pynvml/nvml.py
++++ b/pynvml/nvml.py
+@@ -1475,7 +1475,11 @@ def _LoadNvmlLibrary():
+                             nvmlLib = CDLL(os.path.join(os.getenv("ProgramFiles", "C:/Program Files"), "NVIDIA Corporation/NVSMI/nvml.dll"))
+                     else:
+                         # assume linux
+-                        nvmlLib = CDLL("libnvidia-ml.so.1")
++                        try:
++                            nvmlLib = CDLL("libnvidia-ml.so.1")
++                        except OSError:
++                            # assume NixOS
++                            nvmlLib = CDLL("/run/opengl-driver/lib/libnvidia-ml.so.1")
```

Comment:
This should probably derive from `addOpenGLRunpath.driverLink` rather than hardcoding the path.

Take a look at https://github.com/NixOS/nixpkgs/pull/190023/files#diff-efbc04aeb62d14c6562bd57d0c8f4730a392178d3c1171ffea3731d0f194c2e4R112-R119 for an example of patch templating

Reply:
Thanks, I wasn't aware of that. Updated.

Reply:
You'll also need to replace `/run/opengl-driver/lib` with `@driverLink@/lib` for the substitution to work

Reply:
🤦 I had that change made but didn't stage it for commit.